### PR TITLE
update MI cost estimation

### PR DESCRIPTION
**Pull request by michaellzc**

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3514** <a href="https://app.graphite.dev/github/pr/sourcegraph/handbook/3514" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/handbook/3514?utm_source=stack-comment).

File: content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ The following summarizes the infrastructure costs incurred for a managed instanc
 | $85/mo min. / $0.340/GB        | Data disk, SSD/regional, 250G minimum—exact size depends on customer's repository sizes.                                                                                                                                          |
 | $34/mo                         | Boot disk, SSD/regional, 100G fixed size.                                                                                                                                                                                         |
 | $85/mo / $0.026/GB             | 13 weekly snapshots (taken weekly, retained for 90d)                                                                                                                                                                              |
+| $85/mo / $0.026/GB             | daily snapshots (taken daily, retained for 90d)                                                                                                                                                                                   |
```
Comment (michaellzc) [inline content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:59]: I really can't wrap my head around cost of snapshots.

snapshots are incremental where a new snapshot only contains new and modified data, so ideally the total size of all snapshots shouldn't be too much of different compared to the actual data disk size.

the way we estimate the 13 snaphosts above is

`0.026/Gb * 250Gb * 13 = $85`

but why is it $85 per month?

I don't get it

File: content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md [hunk 2]
```diff
@@ -64,4 +65,24 @@ The following summarizes the infrastructure costs incurred for a managed instanc
...
+
+`db-custom-4-6144`, 100G, $180.26/mo
+
+500-2000 repos
+
+`db-custom-4-10240`, 100G, $200.70/mo
+
+2000+ repos
+
+`db-custom-4-12288`, 100G, $210.92/mo
```
Comment (michaellzc) [inline content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:82]: this is really just unscientific guess, we don't have any real-world data on Cloud SQL yet.

File: content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md [hunk 3]
```diff
@@ -64,4 +65,24 @@ The following summarizes the infrastructure costs incurred for a managed instanc
 | $1.42/GB                       | [Cloud Proxy](https://cloud.google.com/vpc/network-pricing#proxy-instance-charge), 3 are required.                                                                                                                                |
 | $0.12/GB                       | [Internet egress traffic](https://cloud.google.com/vpc/network-pricing#internet_egress) from `us-central1`.                                                                                                                       |
 
+### Cloud SQL usage
+
+Cloud SQL sizing is depending on number of repos
+
+1-500 repos
+
+`db-custom-4-6144`, 100G, $180.26/mo
```
Comment (michaellzc) [inline content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:74]: in docker-compose, we provision 4Gb for `pgsql` and `code-intel` by default.

I suspect customers would use code-intel from day one, and if they don't, the usage will most likely be zero

File: content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md (hunk 1 above)
Comment (filiphaftek) [reply content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:59]: You are right. GCP snpahosts are like in git - additional MB added to previous disks (the diff).  Also the snapshot size is done with used size, not total disk size. I checked also reports and seems like we pay way much less.

**Review by filiphaftek — APPROVED** [review]
👍

File: content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md (hunk 3 above)
Comment (filiphaftek) [reply content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:74]: @rafax any thoughts on CloudSQL pricing?

Comment (rafax) [reply content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:74]: Do you understand how much data we store per 500/1000/2000 repos? How many RAM we need for indices? I'd imagine that we should be able to get away with 2 vCPU instances / 8GB RAM if our dataset is <10GB (and I'd expect <1GB). 

Cloud SQL is ~2x more expensive than underlying compute (per vCPU/RAM), but we should be able to recoup some of that by decreasing MI resource allocation, so this cost sounds reasonable.

Comment (michaellzc) [reply content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:74]: the data we store in the frontend database is pretty minimal and more so depending on users count over repo count.

Some data point

rv2 has 10K repo and 65 users, pgsql uses 1.6GB disk space

glovo has 1K repo and 250 users, pgsql uses 13GB disk space.

When it comes to code-intel things get tricky. If you don't use LSIF, you will have almost 0 usage, however, if you do, things could get scary depending on your retention policy (how long you keep the index around0. Glovo's codeintel db is alsmot 400Gb.

Comment (michaellzc) [reply content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:74]: 😬 also worth mentioning, `rv2` frontend database was still crashing (OOMKILL) at 50Gb memory (I just bumped it to 70G), we never figure out why.

Comment (michaellzc) [reply content/departments/product-engineering/engineering/cloud/devops/managed/cost_estimation.md:74]: the data we store in the frontend database is pretty minimal and more so depending on users count over repo count.

Some data point

`redacted` customer has 10K repo and 65 users, pgsql uses 1.6GB disk space

`redacted` customer has 1K repo and 250 users, pgsql uses 13GB disk space.

When it comes to code-intel things get tricky. If you don't use LSIF, you will have almost 0 usage, however, if you do, things could get scary depending on your retention policy (how long you keep the index around). codeintel db is alsmot 400Gb.

**Pull request closed without merging**